src/core/query/query.js

```diff
@@ -200,19 +190,19 @@ export default class Query {
     }
   }
 
-  /**
-   * Checks if the query is able to be run offline on the local cache.
-   * @return {Boolean} True if it is able to be run offline otherwise false.
-   */
   isSupportedOffline() {
-    let supported = true;
+    return Object.keys(this.filter).reduce((supported, key) => {
```

Comment:
`findKey()` can take a predicate as a scond argument, think this would simplify things. This looks much more complicated than before, and again shouldn't be the focus of a PR which merges repos. Is something else going on here during the merge? The longer the merge takes, the longer we have to freeze code.

Reply:
I was setting up unit tests and I chose to use the Query and Aggregation unit tests as a way to figure out how things are setup. Part of further PRs will be to remove the use of [loadsh](https://lodash.com) as it is the main source as to why our bundles are somewhat big in size. Many of the lodash functions can be replaced by plain JavaScript functions.

With that being said, yes this probably shouldn't be part of this PR but we will do the this in the future so I rather leave it now.